notebooks/samples/explanations/ai-explanations-image-tf2.ipynb

```diff
@@ -0,0 +1,1758 @@
+{
```

Comment:
<h2>Export the model</h2><p> </p><p>When using TensorFlow 2.x, you export the model as a <code>SavedModel</code>  and load it into Cloud Storage. During export, you need to define a serving function to convert data to the format your model expects. If you send encoded data to AI Platform, your serving function ensures that the data is decoded on the model server before it is passed as input to your model. </p><h3><br></h3><h3>Serving function for image data</h3><p> </p><p>Sending base 64 encoded image data to AI Platform is more space efficient. Since this deployed model expects input data as raw bytes, you need to ensure that the b64 encoded data gets converted back to raw bytes before it is passed as input to the deployed model.</p><p> </p><p>To resolve this, define a serving function (<code>serving_fn</code>) and attach it to the model as a preprocessing step. Add a <code>@tf.function</code> decorator so the serving function is part of the model's graph (instead of upstream on a CPU). </p><p> </p><p>When you send a prediction or explanation request, the request goes to the serving function (<code>serving_fn</code>), which preprocesses the b64 bytes into raw numpy bytes (<code>preprocess_fn</code>). At this point, the data can be passed to the model (<code>m_call</code>).</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/ai-platform-samples/pull/192/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='43'/>